python/private/pypi/dependency_specifier_flag.bzl

```diff
@@ -0,0 +1,97 @@
+load("//python/private:toolchain_types.bzl", "TARGET_TOOLCHAIN_TYPE")
+load(":pep508_evaluate.bzl", "evaluate")
+
+def _impl(ctx):
+    env = {}
+
+    runtime = ctx.toolchains[TARGET_TOOLCHAIN_TYPE].py3_runtime
+    if runtime.interpreter_version_info:
+        version_info = runtime.interpreter_version_info
+        env["python_version"] = "{v.major}.{v.minor}".format(v = version_info)
+        full_version = format_full_version(version_info)
+        env["python_full_version"] = full_version
+        env["implementation_version"] = full_version
+    else:
+        env["python_version"] = get_flag(ctx.attr._python_version)
+        full_version = get_flag(ctx.attr._python_full_version)
+        env["python_full_version"] = full_version
+        env["implementation_version"] = full_version
+
+    # We assume cpython if the toolchain doesn't specify because it's most
+    # likely to be true.
+    env["implementation_name"] = runtime.implementation_name or "cpython"
+
+    # todo: map from os constraint
+    env["os_name"] = struct()
+
+    # todo: map from os constraint
+    env["sys_platform"] = struct()
+
+    # todo: map from cpu flag (x86_64, etc)
+    env["platform_machine"] = struct()
+
+    # todo: add PyRuntimeInfo.platform_python_implementation
+    # The values are slightly different to implementation_name
+    env["platform_python_implementation"] = runtime.implementation_name
+
+    # todo: add flag to carry this
+    env["platform_release"] = struct()
+
+    # todo: map from os constraint
+    env["platform_system"] = struct()
+
+    # todo: add flag to carry this
+    env["platform_version"] = struct()
```

Comment:
A note to myself. Right now `pypi/config_settings` is generating something similar for `osx_version` and maybe this flag could be used instead of creating a new separate flag for the `osx_version`.